framework/src/engine/generator/network_endpoint.ts

```
@@ -226,19 +226,44 @@ export class NetworkEndpoint extends BaseNetworkEndpoint {
 			);
 
 			try {
-				for (const transactionBytes of transactionsData.transactions) {
-					const transaction = Transaction.fromBytes(transactionBytes);
-					await this._receiveTransaction(transaction);
-				}
+				await Promise.all(
+					transactionsData.transactions.map(async transactionBytes => {
+						const transaction = Transaction.fromBytes(transactionBytes);
```

this will throw instead of returning undefined. Also, try-catch can be bundled.